src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIAssistantChatClient.cs

```diff
@@ -203,18 +203,53 @@ public async IAsyncEnumerable<ChatResponseUpdate> GetStreamingResponseAsync(
                     break;
 
                 case MessageContentUpdate mcu:
-                    yield return new(mcu.Role == MessageRole.User ? ChatRole.User : ChatRole.Assistant, mcu.Text)
+                    ChatResponseUpdate textUpdate = new(mcu.Role == MessageRole.User ? ChatRole.User : ChatRole.Assistant, mcu.Text)
                     {
                         AuthorName = _assistantId,
                         ConversationId = threadId,
                         MessageId = responseId,
                         RawRepresentation = mcu,
                         ResponseId = responseId,
                     };
+
+                    // Add any annotations from the text update. The OpenAI Assistants API does not support passing these back
+                    // into the model (MessageContent.FromXx does not support providing annotations), so they end up being one way and are dropped
+                    // on subsequent requests.
+                    if (mcu.TextAnnotation is { } tau)
+                    {
+                        string? fileId = null;
+                        string? toolName = null;
+                        if (!string.IsNullOrWhiteSpace(tau.InputFileId))
+                        {
+                            fileId = tau.InputFileId;
+                            toolName = "file_search";
```

Comment:
Just curious, what do you expect developers to do with this tool name information? If we chose not to set it in these cases, would there be any drawback?

Asking because it feels like annotations could have file IDs even if they came from other tools, but maybe that's a problem for the future and not for now.

Reply:
Just to be able to distinguish the source of the file ID, if it matters. InputFileId is documented to come from file_search.